cli/src/commands/token/create/dev.rs

```diff
@@ -99,80 +99,83 @@ async fn read_config(
-		// mode will be running locally
-		if let Some(ports) = matchmaker.docker.as_ref().and_then(|x| x.ports.as_ref()) {
-			for (label, port) in ports {
-				lobby_ports.insert(label.clone(), port);
+			// Global overrides take priority because we don't know what game
+			// mode will be running locally
+			if let Some(ports) = matchmaker.docker.as_ref().and_then(|x| x.ports.as_ref()) {
+				for (label, port) in ports {
+					lobby_ports.insert(label.clone(), port);
+				}
 			}
-		}
-
-		lobby_ports
-	};
-
-	let dev_ports = dev_ports
-		.into_iter()
-		.map(|(label, port_config)| {
-			//
-			let (port, port_range) = if let Some(dev_port) = port_config.dev_port {
-				(Some(dev_port), None)
-			} else if let Some(dev_port_range) = port_config.dev_port_range.as_ref() {
-				(None, Some(dev_port_range.clone()))
-			} else if let Some(port) = port_config.port {
-				(Some(port), None)
-			} else if let Some(port_range) = port_config.port_range.as_ref() {
-				(None, Some(port_range.clone()))
-			} else {
-				bail!("missing both port and port_range")
-			};
-
-			Ok((
-				label,
-				models::CloudMatchmakerDevelopmentPort {
-					port,
-					port_range,
-					protocol: port_config
-						.dev_protocol
-						// Default to non-TLS version of the given protocol
-						.unwrap_or({
-							use models::CloudVersionMatchmakerPortProtocol::*;
-							match port_config.protocol {
-								Some(Https) | Some(Http) | None => Http,
-								Some(Tcp) | Some(TcpTls) => Tcp,
-								Some(Udp) => Udp,
-							}
-						}),
-				},
-			))
-		})
-		.collect::<GlobalResult<HashMap<_, _>>>()?;
-
-	Ok((dev_hostname, dev_ports))
+
+			lobby_ports
+		};
+
+		let dev_ports = dev_ports
+			.into_iter()
+			.map(|(label, port_config)| {
+				//
+				let (port, port_range) = if let Some(dev_port) = port_config.dev_port {
+					(Some(dev_port), None)
+				} else if let Some(dev_port_range) = port_config.dev_port_range.as_ref() {
+					(None, Some(dev_port_range.clone()))
+				} else if let Some(port) = port_config.port {
+					(Some(port), None)
+				} else if let Some(port_range) = port_config.port_range.as_ref() {
+					(None, Some(port_range.clone()))
+				} else {
+					bail!("missing both port and port_range")
+				};
+
+				Ok((
+					label,
+					models::CloudMatchmakerDevelopmentPort {
+						port,
+						port_range,
+						protocol: port_config
+							.dev_protocol
+							// Default to non-TLS version of the given protocol
+							.unwrap_or({
+								use models::CloudVersionMatchmakerPortProtocol::*;
+								match port_config.protocol {
+									Some(Https) | Some(Http) | None => Http,
+									Some(Tcp) | Some(TcpTls) => Tcp,
+									Some(Udp) => Udp,
+								}
+							}),
+					},
+				))
+			})
+			.collect::<GlobalResult<HashMap<_, _>>>()?;
+
+		return Ok((dev_hostname, dev_ports));
+	}
+
+	Ok((
+		"127.0.0.1".to_string(),
+		HashMap::<String, models::CloudMatchmakerDevelopmentPort>::new(),
+	))
```

Comment:
@NathanFlurry this was the intention of what to return right?

Reply:
Yes